svgdigitizer/svgplot.py

```diff
@@ -1656,6 +1656,68 @@ def _min_real_root(cls, polynomial, tmin, tmax):
 
         return min(eligible_roots)
 
+    @property
+    def schema(self):
+        """A frictionless `Schema` object, including a `Fields` object
+        describing the voltage and current axis of the original plot
+        including original units.
+
+        EXAMPLES::
+
+            >>> from svgdigitizer.svg import SVG
+            >>> from io import StringIO
+            >>> svg = SVG(StringIO(r'''
+            ... <svg>
+            ...   <g>
+            ...     <path d="M 0 100 L 100 0" />
+            ...     <text x="0" y="0">curve: 0</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M 0 200 L 0 100" />
+            ...     <text x="0" y="200">t1: 0</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M 100 200 L 100 100" />
+            ...     <text x="100" y="200">t2: 1</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 100 L 0 100" />
+            ...     <text x="-100" y="100">y1: 0</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 0 L 0 0" />
+            ...     <text x="-100" y="0">y2: 1</text>
+            ...   </g>
+            ... </svg>'''))
+            >>> plot = SVGPlot(svg)
+            >>> plot.schema  # doctest: +NORMALIZE_WHITESPACE
+            {'fields': [{'name': 't', 'unit': None, 'orientation': 'x'},
+                        {'name': 'y', 'unit': None, 'orientation': 'y'}]}
+
+
+        """
+
+        def orientation(axis):
+            if axis == "vertical":
+                return "y"
+            if axis == "horizontal":
+                return "x"
+
+            raise Exception("The axis must be either 'vertical' or 'horizontal")
+
+        fields = []
+        for key, label in self.axis_orientations.items():
+            field = {
+                "name": label,
+                "unit": self.axis_labels[label],
+                "orientation": orientation(key.value),
+            }
+            fields.append(field)
+
+        from frictionless import Schema
+
+        return Schema(fields=fields)
```

Comment:
```suggestion
        from frictionless import Schema

        orientations = {
            "vertical": "y",
            "horizontal": "x",
        }

        return Schema(fields={
            "name": label,
            "unit": self.axis_labels[label],
            "orientation": orientations[key.value],
        } for key, label in self.axis_orientations.items())
```

I think you can just do a dict lookup instead of implementing a function. A key error is going to be slightly more cryptic but it cannot happen unless we have a bug in the program anyway.